diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -374,9 +374,12 @@ namespace diff_drive_controller{
     config.publish_rate = publish_rate;
     config.enable_odom_tf = enable_odom_tf_;
 
-    dyn_reconf_server_ = std::make_shared<ReconfigureServer>(controller_nh);
-    dyn_reconf_server_->updateConfig(config);
+    dyn_reconf_server_ = boost::make_shared<ReconfigureServer>(dyn_reconf_server_mutex_, controller_nh);
     dyn_reconf_server_->setCallback(boost::bind(&DiffDriveController::reconfCallback, this, _1, _2));
+    // Update parameters
+    dyn_reconf_server_mutex_.lock();
+    dyn_reconf_server_->updateConfig(config);
+    dyn_reconf_server_mutex_.unlock();
```

Comment:
Didn't notice this before, but let's make sure we call `updateConfig` before `setCallback`. Otherwise, the callback will be triggered when we update the config.

Reply:
Fixed